scalding-core/src/main/scala/com/twitter/scalding/CancellationHandler.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.scalding
+
+import scala.concurrent.{ Future, ExecutionContext => ConcurrentExecutionContext, Promise }
+
+trait CancellationHandler { outer =>
+  def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit]
+  def compose(other: CancellationHandler): CancellationHandler = new CancellationHandler {
+    override def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = {
+      other.stop().flatMap(_ => outer.stop())
+    }
+  }
+  def compose(other: Future[CancellationHandler]): CancellationHandler = new CancellationHandler {
+    override def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = {
+      other.flatMap(_.stop).flatMap(_ => outer.stop())
+    }
+  }
+}
+
+object CancellationHandler {
+  def empty: CancellationHandler = new CancellationHandler {
+    def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = Future(())
+  }
+
+  def cancellable[T](f: Future[T], onCancel: => Unit = ())(implicit ec: ConcurrentExecutionContext): (Future[T], CancellationHandler) = {
+    val p = Promise[T]
+    val first = Future.firstCompletedOf(List(p.future, f))
+    val cancel = new CancellationHandler {
+      override def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = Future {
+        first.onFailure { case e => onCancel }
```

Comment:
I don't see why we need this. Can you add a comment?

Reply:
i initially thought that maybe we would like to stop a Flow as part of a callback on the failed future. but i don't think this is really needed currently. although, tbh i'm not sure if i should even bother with wrapping Futures with this function, as we aren't really able to cancel the execution. lmk if you think it's worth keeping this around.

Reply:
my only question is whether we want to wrap `fromFuture` with this so that the downstreams can be failed without waiting.